libs/core/langchain_core/tools.py

```diff
@@ -1218,3 +1218,32 @@ class BaseToolkit(BaseModel, ABC):
     @abstractmethod
     def get_tools(self) -> List[BaseTool]:
         """Get the tools in the toolkit."""
+
+
+def _get_description_from_runnable(runnable: Runnable) -> str:
+    """Generate a placeholder description of a runnable."""
+    input_schema = runnable.input_schema.schema()
+    return f"Takes {input_schema}."
+
+
+def _get_args_schema_for_runnable(runnable: Runnable) -> Type[BaseModel]:
+    """Get object schema from runnable input schema."""
+    schema_name = f"{runnable.get_name()}Schema"
+    input_name = "input"
+    fields = {input_name: (runnable.InputType, Field(...))}
+    return create_model(schema_name, **fields)  # type: ignore
+
+
+def convert_runnable_to_tool(
```

Comment:
One odd thing is that we lose any optimizations from the runnables that might exist (e.g., batch, stream) etc.

Do we need a more direct way to promote a runnable to a tool, and have the structured tool delegate to the underlying methods?

Reply:
+ on this. would it be better to create a `StructuredTool.from_runnable` and add overrides for batch/stream in the event `self._runnable` instead of `self.func` is set?

Reply:
think this is probably an issue with tools that can be handled separately (eg StructuredTool.from_function should take batch/stream variants)